astroid/inference.py

```diff
@@ -45,11 +54,62 @@
     NameInferenceError,
     _NonDeducibleTypeHierarchy,
 )
+from astroid.helpers import (
+    class_instance_as_index,
+    is_subtype,
+    is_supertype,
+    object_type,
+    safe_infer,
+)
 from astroid.interpreter import dunder_lookup
+from astroid.manager import AstroidManager
+from astroid.node_classes import (
+    Arguments,
+    AssignAttr,
```

Comment:
I'm not sure this is really an improvement here. Readability wise I would prefer the old version of `astroid/inference.py`. Especially for `nodes`, `decorators`, `helpers`, `protocols`, and `utils`.

Reply:
The thing is import by entire module look nicer but they also hide circular imports. Files are enormous so we want to use import by module so it feels reasonable, but I think we also need to burst those enormous 5000+ lines file importing 50+ classes from a single file, so we have manageable import in the results and we can see where the circular imports are. There are global values hidden because of that (Thing like that are not a one time thing: https://github.com/PyCQA/astroid/pull/972/files). Maybe we could start by splitting file first ?

Reply:
Let's do everything except `nodes`. It does have `__all__` so I think it should be fine (for now).

Reply:
I'm going to split this MR into multiple one, there is too much in it right now.

Reply:
I removed this part, it did not help for the cyclic import at all in facT.